examples/pebbleui/js/modules/PebbleUI.js

```diff
@@ -0,0 +1,561 @@
+var PebbleUI = function(rocky, options) {
+  options = options || {};
+
+  this._windows = [];
+
+  // Load the default font
+  this._defaultFont = options.defaultFont || rocky.FONT_KEY_GOTHIC_18;
+
+  this._getTopWindow = function() {
+    // short circuit if there's nothing to _render
+    if (!this._windows || this._windows.length < 1) return null;
+
+    // otherwise _render the active screen
+    return this._windows[this._windows.length - 1];
+  };
+
+  this._dirty = false;
+
+  this.mark_dirty = function() {
+    // don't do anything if it's already marked dirty
+    if (this._dirty === true) return;
+
+    // Otherwise, render when possible
+    this._dirty = true;
+    setTimeout(function() {
+      rocky.mark_dirty();
+      this._dirty = false;
+    }.bind(this), 0);
+  };
+
+  this._render = function(ctx, bounds) {
+    var win = this._getTopWindow();
+    if (win) win._render(ctx, bounds);
+  };
+
+  var parent = this;
+
+  this._window = window || {};
```

Comment:
`|| {}` here doesn't prevent you from `addEventListener is not a function` a few lines later.